tc/sgx/trusted_worker_manager/enclave_wrapper/file_io_processor.cpp

```diff
@@ -0,0 +1,298 @@
+/* Copyright 2019 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <string>
+#include <fstream>
+#include <iostream>
+#include <string.h>
+
+#include "file_io_processor.h"
+#include "log.h"
+#include "tcf_error.h"
+
+#define SUCCESS 0
+#define FAILED -1
+
+using namespace std;
+
+// Opens file and updates status SUCCESS or FAILED in result parameter
+uint32_t FileOpen(string fileName, uint8_t *result, size_t resultSize) {
+    string resultStr;
+    if (fileName.empty()) {
```

Comment:
The filename is empty test code is duplicated in all the functions. Please write a local static function to test this and call from the other functions to reduce duplicated code.

Reply:
Done